newrelic/data_source_newrelic_synthetics_monitor.go

```diff
@@ -0,0 +1,83 @@
+package newrelic
+
+import (
+	"fmt"
+	"log"
+
+	synthetics "github.com/dollarshaveclub/new-relic-synthetics-go"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func dataSourceNewRelicSyntheticsMonitor() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceNewRelicSyntheticsMonitorRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"api_key": {
+				Type:        schema.TypeString,
+				Required:    true,
+				DefaultFunc: schema.EnvDefaultFunc("NEWRELIC_API_KEY", nil),
+				Sensitive:   true,
+			},
+			"max_check": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Default:  1000,
+			},
+		},
+	}
+}
+
+func dataSourceNewRelicSyntheticsMonitorRead(d *schema.ResourceData, meta interface{}) error {
+
+	apiKey := d.Get("api_key").(string)
+	maxCheck := d.Get("max_check").(int)
+
+	conf := func(s *synthetics.Client) {
+		s.APIKey = apiKey
+	}
+
+	syntheticsClient, _ := synthetics.NewClient(conf)
+
+	log.Printf("[INFO] Reading New Relic synthetics monitors")
+
+	offset := 0
+	max := 100
+	monitors, err := syntheticsClient.GetAllMonitors(uint(offset), uint(max))
+	var monitor *synthetics.ExtendedMonitor
+	name := d.Get("name").(string)
+	for offset <= maxCheck {
+		if len(monitors.Monitors) > 0 && err == nil {
+			mon := *monitors
+
+			for i := 0; i < len(monitors.Monitors); i++ {
+				if mon.Monitors[i].Name == name {
+					monitor = mon.Monitors[i]
+					break
+				}
+			}
+		} else {
+			break
+		}
+
+		offset = offset + 100
+		monitors, err = syntheticsClient.GetAllMonitors(uint(offset), uint(max))
+	}
+	if err != nil {
+		return err
+	}
+
+	if monitor == nil {
+		return fmt.Errorf("The name '%s' does not match any New Relic monitors.", name)
+	}
+
+	d.SetId(monitor.ID)
+	d.Set("name", monitor.Name)
+	d.Set("monitor_id", monitor.ID)
```

Comment:
This needs to be defined in the schema.

Reply:
@paultyng You want people to provide the monitor id? Doesnt it make more sense to get that programmatically?

Reply:
Sorry, should have been more clear, in the schema you can specify `Computed` (and not `Optional` or `Required`) for data that is only output.

Reply:
https://github.com/hashicorp/terraform/blob/0cc9e050ecd4a46ba6448758c2edc0b29bef5695/helper/schema/schema.go#L109-L110

Reply:
https://www.terraform.io/docs/extend/schemas/schema-behaviors.html#computed

(these are some new docs we launched recently!)

Reply:
Ahh thanks that makes sense! Will do